lib/ws-channel.js

```diff
@@ -554,15 +586,17 @@ WebsocketChannel.prototype._fatalError = function(err) {
     err = new Error(err);
   debug('detach on fatal error: %s', err.message);
   this._detach();
+  this._closeWebsocket();
+  this._closed = err.message;
```

Comment:
What about storing the actual `Error` here so that we can preserve a stacktrace closer to the origin?